src/python/gem5/resources/resource.py

```diff
@@ -554,6 +556,77 @@ def _get_weights_and_simpoints_from_file(
         return simpoint_list, weight_list
 
 
+class SuiteResource(AbstractResource):
+    def __init__(
+        self,
+        workloads: List[Dict[str, Any]],
+        local_path: Optional[str] = None,
+        resource_version: Optional[str] = None,
+        description: Optional[str] = None,
+        source: Optional[str] = None,
+    ) -> None:
+        super().__init__(
+            local_path=local_path,
+            description=description,
+            source=source,
+            resource_version=resource_version,
+        )
+        print("in constructor")
+        self.resource_version = resource_version
+        self.workloads = workloads
+        self.workload_obj_list = []
+        self.index = 0
+        self._local_path = local_path
+        self._description = description
+        self._source = source
+        self._obtain_workloads_list()
+
+    def _obtain_workloads_list(self) -> None:
+        for workload in self.workloads:
+            self.workload_obj_list.append(
+                Workload(
+                    workload["id"],
+                    resource_version=workload["resource_version"],
+                )
+            )
+
+    def __iter__(self) -> "SuiteResource":
+        return self
+
+    def __next__(self) -> "Workload":
+        try:
+            workload = self.workload_obj_list[self.index]
+        except IndexError:
+            raise StopIteration
+        self.index += 1
+        return workload
```

Comment:
As previously suggested, I think you can do this:
```suggestion
    def __iter__(self) -> "SuiteResource":
        for workload in self.workload_obj_list:
            yield workload
```

Reply:
This change is done, we haven't pushed it yet, we checked and it works the same way.